roadmap/implementers-guide/src/node/availability/availability-recovery.md

```diff
@@ -20,3 +22,148 @@ Output:
 - AvailabilityStore::QueryChunk
 
 ## Functionality
+
+We hold a state which tracks the current recovery interactions we have live, as well as which request IDs correspond to which interactions. An interaction is a structure encapsulating all interaction with the network necessary to recover the available data.
+
+```rust
+type ChunkResponse = Result<(PeerId, ErasureChunk), Unavailable>;
+
+struct AwaitedChunk {
+    issued_at: Instant,
+    validator_index: ValidatorIndex,
+    candidate_hash: CandidateHash,
+    response: ResponseChannel<ChunkResponse>,
+}
+
+struct State {
+    /// Each interaction is implemented as its own async task, and these handles are for communicating with them.
+    interactions: Map<CandidateHash, InteractionHandle>,
+    /// A recent block hash for which state should be available.
+    live_block_hash: Hash,
```

Comment:
Probably this should be a `BlockNumber` or at least we should check that when we update this field the new block is higher than the old one?

Reply:
What we really care about is `SlotNumber`, not block number. But yeah, there is some risk of moving backwards.